catkin_tools/resultspace.py

```diff
@@ -59,6 +59,7 @@ def get_resultspace_environment(result_space_path, quiet=False, cached=True):
     # Use fallback shell if using a non-standard shell
     if shell_name not in ['bash', 'zsh']:
         shell_name = 'bash'
+        shell_path = "/bin/bash"
```

Comment:
Actually, specifying the default not here, but above in `shell_path = os.environ.get('SHELL', '/bin/bash')` would allow to set `SHELL` to something custom not ending in `/bash` and still have it work (with the `--norc` flag as if it were bash). With this PR, the value in `SHELL` gets overwritten if it does not end in `/bash` or `/zsh`.